openwisp_controller/connection/api/serializers.py

```diff
@@ -31,6 +32,22 @@ class CommandSerializer(ValidatedDeviceFieldSerializer):
         pk_field=serializers.UUIDField(format='hex_verbose'),
     )
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # show only connections and command types available for the device
+        if device_id := self.context.get('device_id'):
+            self.fields['connection'].queryset = self.fields[
+                'connection'
+            ].queryset.filter(device_id=device_id)
+            device = Device.objects.only('organization_id', 'id').get(pk=device_id)
+            # filter command types based on the device's organization
+            allowed_commands = Command.get_org_choices(device.organization_id)
+            # this translates the commands to their labels
+            commands_map = dict(COMMAND_CHOICES)
+            self.fields['type'].choices = [
+                (i, commands_map[i]) for i in commands_map if i in allowed_commands
+            ]
```

Comment:
Can we move this to a class method in Command model?

Reply:
Yes. Makes sense

Reply:
Should i combine this with `get_org_choices` as they seem to to do the same thing?

Reply:
I think, that is an acceptable solution. I am afraid, that will be backward incompatible though. 

We would also need to update this code

https://github.com/openwisp/openwisp-controller/blob/523f4dab1b84d4b0611085ec1fab60aeb65e7470/openwisp_controller/connection/base/models.py#L473-L475


cc @nemesifier

Reply:
Yes, i think this is the only place where it is used, so it should be easier now to do this change. Should I proceed?

Reply:
Yes, go ahead!